### PR TITLE
Merchant Dashboard Complete - needs 2 model tests

### PR DESCRIPTION
**Pull request by haewonito**
here goes nothing.

### CODE REVIEW

File: app/views/merchant_dashboards/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+</div>
+
+
+<div id='items-ready-to-ship'>
+<h4>Items Ready to Ship</h4>
+<% @merchant.items_ready_to_ship.each do |data| %>
+  <p>
+  <%= data.item_name %> - Invoice #<%= link_to "#{data.invoice_id}", "/merchants/#{@merchant.id}/invoices/#{data.invoice_id}" %> - <%= data.invoice_created_at.strftime("%A, %B %-d, %Y") %>
+  </p>
+<% end %>
```
Comment (michab17) [inline app/views/merchant_dashboards/index.html.erb:24]: What is data referring to here, is it an invoice item?

File: spec/features/merchant_dashboards/index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      @transaction_5 = @invoice_3.transactions.create(credit_card_number: 1234123412341234, credit_card_expiration_date: '2012-03-27', result: 'success')
+      @transaction_6 = @invoice_4.transactions.create(credit_card_number: 1234123412341234, credit_card_expiration_date: '2012-03-27', result: 'success')
+      @transaction_7 = @invoice_4.transactions.create(credit_card_number: 1234123412341234, credit_card_expiration_date: '2012-03-27', result: 'success')
+      @transaction_8 = @invoice_5.transactions.create(credit_card_number: 1234123412341234, credit_card_expiration_date: '2012-03-27', result: 'success')
+      @transaction_9 = @invoice_5.transactions.create(credit_card_number: 1234123412341234, credit_card_expiration_date: '2012-03-27', result: 'success')
+      @transaction_10 = @invoice_5.transactions.create(credit_card_number: 1234123412341234, credit_card_expiration_date: '2012-03-27', result: 'success')
+      @transaction_11 = @invoice_6.transactions.create(credit_card_number: 1234123412341234, credit_card_expiration_date: '2012-03-27', result: 'failed')
```
Comment (michab17) [inline spec/features/merchant_dashboards/index_spec.rb:46]: Maybe we should look into factory bot 😂

File: spec/features/merchant_dashboards/index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+      visit "/merchants/#{@merchant.id}/dashboard"
```
Comment (michab17) [inline spec/features/merchant_dashboards/index_spec.rb:48]: Do we want to use merchant_dashboard_path for this?

File: spec/features/merchant_dashboards/index_spec.rb [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+    it "I see each invoice id as a link to my merchant's invoice show page" do
+      within ("div#items-ready-to-ship") do
+        click_on "#{@invoice_6.id}"
+        expect(current_path).to eq("/merchants/#{@merchant.id}/invoices/#{@invoice_6.id}")
+
+        visit "/merchants/#{@merchant.id}/dashboard"
+
+        click_on "#{@invoice_7.id}"
+        expect(current_path).to eq("/merchants/#{@merchant.id}/invoices/#{@invoice_7.id}")
```
Comment (michab17) [inline spec/features/merchant_dashboards/index_spec.rb:100]: I dont know if we would need to test both of the links, if one works the other one should work as well

File: spec/features/merchant_dashboards/index_spec.rb [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+    it "I see date that the invoice was created, formatted 'Monday, July 18, 2019'" do
+      date = @invoice_6.created_at.strftime("%A, %B %-d, %Y")
+      expect(page).to have_content(date)
+    end
+
+    it "I see the item list ordered from oldest to newest" do
+      within ("div#items-ready-to-ship") do
+        expect("#{@invoice_6.id}").to appear_before("#{@invoice_7.id}")
+      end
```
Comment (michab17) [inline spec/features/merchant_dashboards/index_spec.rb:112]: Very nice we couldnt get this test to work

**Review by michab17 — COMMENTED** [review]
Added some comments, pretty much everything looks great!

File: spec/features/merchant_dashboards/index_spec.rb (hunk 2 above)
Comment (samueldevine) [reply spec/features/merchant_dashboards/index_spec.rb:46]: agreed

File: spec/features/merchant_dashboards/index_spec.rb (hunk 3 above)
Comment (samueldevine) [reply spec/features/merchant_dashboards/index_spec.rb:48]: yeah i think that's a good call. we don't have resourceful routing setup yet but will add it in soon

File: app/views/merchant_dashboards/index.html.erb (hunk 1 above)
Comment (samueldevine) [reply app/views/merchant_dashboards/index.html.erb:24]: its a customer AR object we created in the method - similar to yours it's not referring to a single object so I wasn't sure what to call it. data has the name of an item, the invoice id for that item, and the invoice created_at

File: spec/features/merchant_dashboards/index_spec.rb (hunk 2 above)
Comment (samueldevine) [reply spec/features/merchant_dashboards/index_spec.rb:46]: lol

**Pull request merged**